### ISSUE TITLE
Uploading resized images

### ISSUE THREAD

**Issue by NethInt** [deliciousbrains/wp-amazon-s3-and-cloudfront#463]
Hello,

We are using S3 offload pro version. Before S3 offload plugin, we have used 3 default sizes for the images (Thumbnail size 150 x 150, Medium size 300 x 300, Large size 1024 x 1024). Apart from that we have resized our images for responsive reasons. So there are images like 360x225, 100x72, 400x300...

First we have installed S3 offload and copied files using plugin settings. Once copying is finished we have noticed that only the original image and default sizes are copied to S3. Then we have created new add_image_size in function.php and uploaded new image. It's working fine. Copying to S3 with default sizes and newly added sizes. However when we try to re-sync old files from S3, it's the same. Newly defined sizes are in the content folder, yet not uploading to S3 bucket. 

Thank you.

**Comment by willfre** [comment]
Similar issue for me, my theme generates custom sizes and they aren't being uploaded.

No response from dev?

**Comment by NethInt** [comment]
Well got no response from dev. But found a walk around solution. 

Our problem was that additional customs sizes (360x225, 100x72, 400x300...) for images were generated only when we need. By resizing original image via theme function. But there was no mapping for those custom size images in Wordpress database, indicating that this original image has these custom sizes. 
When we add new add_image_size function it create database entries for the new image uploads with new sizes. **So no problem for the newly uploaded images.** However for old images still doesn't have that 'mapping in database'.

If it's possible for you to re-upload old images, you will be able sync all the images with custom sizes without any issue.

However we had like 19000 old images. So we found a solution to add the mapping in database with another plugin. We have regenerated old images with custom sizes, and it created the mapping that original image has these kind of custom sizes. This is the link for the plugin https://wordpress.org/plugins/regenerate-thumbnails/
 
So bottom line is that the S3 Offload plugin only sync the resized images when there is a database entry. If you can re-upload old images after add_image_size. Otherwise regenerate custom sizes to update database mappings.

**Comment by ianmjones** [comment]
On the fly image cropping is the Devil's work!

https://deliciousbrains.com/on-the-fly-image-processing/

Once you've got images sizes properly registered using something like the Regenerate Thumbnails plugin usually works to copy newly generated thumbnails up to S3.

@NethInt If you have a Pro license please contact us via the plugin's built in support or directly via email! GitHub issues are for reporting bugs or raising feature requests for the Lite plugin, you're entitled to our priority email support if you have a license, we strive to provide top-notch support via email.

For Lite plugin support, please raise a request at https://wordpress.org/support/plugin/amazon-s3-and-cloudfront.
